azure/services/scalesets/scalesets.go

```diff
@@ -673,27 +578,57 @@ func (s *Service) getVirtualMachineScaleSetDefaultNetworkConfiguration(vmssSpec
 				})
 		}
 	}
-	return &[]compute.VirtualMachineScaleSetNetworkConfiguration{{
-		Name: pointer.String(vmssSpec.Name),
-		VirtualMachineScaleSetNetworkConfigurationProperties: &compute.VirtualMachineScaleSetNetworkConfigurationProperties{
-			Primary:            pointer.Bool(true),
-			EnableIPForwarding: pointer.Bool(true),
-			IPConfigurations: &[]compute.VirtualMachineScaleSetIPConfiguration{
-				{
-					Name: pointer.String(vmssSpec.Name),
-					VirtualMachineScaleSetIPConfigurationProperties: &compute.VirtualMachineScaleSetIPConfigurationProperties{
-						Subnet: &compute.APIEntityReference{
-							ID: pointer.String(azure.SubnetID(s.Scope.SubscriptionID(), vmssSpec.VNetResourceGroup, vmssSpec.VNetName, vmssSpec.SubnetName)),
-						},
-						Primary:                         pointer.Bool(true),
-						PrivateIPAddressVersion:         compute.IPVersionIPv4,
-						LoadBalancerBackendAddressPools: &backendAddressPools,
+	nicConfigs := []compute.VirtualMachineScaleSetNetworkConfiguration{}
+	for i, n := range vmssSpec.NetworkInterfaces {
```

Comment:
Unless I'm mistaken: if `len(vmssSpec.NetworkInterfaces) == 0` this would not add any network configuration. Wondering how to handle this best. To keep code here simple, should we maybe add `networkInterfaces` to the defaulter webhook if none are specified?

Reply:
Correct, the defaulting in the webhook is already being done so there should never be a case where `len(vmssSpec.NetworkInterfaces) == 0`:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/exp/api/v1beta1/azuremachinepool_default.go#L136

Reply:
hm, true. Wonder why this didn't catch when we upgraded CAPZ.. But maybe there were other issues. Looks fine like this.

Reply:
What happens when the `amp.Spec.Template.SubnetName` and `amp.Spec.Template.AcceleratedNetworking` is not defined in the spec, will we be setting empty subnetName and dereferencing an empty pointer in the [defaulting webhook](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/2d7ffd3ad1db2c57171337a1fc2aa276b4d2e0e6/exp/api/v1beta1/azuremachinepool_default.go#L136-L145)?

Reply:
> and dereferencing an empty pointer 

where do you see dereferencing of a pointer in the webhook?

If accelerated networking is nil we use the value of the SKU to determine if it's supported https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3188/files#diff-a6779b0b37d0dcdbe7dcd6f00150a22e12ac067a96269038452d27ba39349a90R603 

For empty subnet name it won't be empty because it's defaulted to the Cluster subnet name if empty before it reaches this part of the code on every reconcile loop: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/2d7ffd3ad1db2c57171337a1fc2aa276b4d2e0e6/azure/scope/machinepool.go#L726

Reply:
> > and dereferencing an empty pointer
> 
> where do you see dereferencing of a pointer in the webhook?

Sorry my bad, didn't post the question right. 

My other concerns are addressed too. Thanks for pinning the info!

/lgtm